src/file-selector.ts

```diff
@@ -108,34 +108,51 @@ function flatten<T>(items: any[]): T[] {
 async function fromDataTransferItem(
   item: DataTransferItem,
   entry?: FileSystemEntry | null,
-) {
-  // Check if we're in a secure context; due to a bug in Chrome (as far as we know) the browser crashes when calling this API (yet to be confirmed as a consistent behavior).
-  //
+): Promise<FileWithPath> {
+  const fileWithPath = await fromFileSystemHandle(item, entry);
+
+  if (fileWithPath) {
+    return fileWithPath;
+  }
+
+  const file = item.getAsFile();
+
+  if (!file) {
+    throw new Error("Transferred item is not a file.");
+  }
+
+  return toFileWithPath(file, entry?.fullPath);
+}
+
+async function fromFileSystemHandle(
+  item: DataTransferItem,
+  entry?: FileSystemEntry | null,
+): Promise<File | null> {
+  // Check if we're in a secure context; otherwise due to a bug in Chrome the browser can crash when calling this API.
```

Comment:
I mean, it has no functional value, but the original message change wasn't necessary 😉

Reply:
Fair, I will restore the original. I think I changed it because it did not make sense in some initial refactoring I did, but it's pretty much the same after re-basing.